Creacion del CRUD para crear, obtener, borrar y editar los Goals por cada usuario, adicional se obtiene el progreso para alcanzar el goal

jlnunezg10 opened the pull request:
Hola chicos en el siguiente link de Trello pueden observar las rutas creadas con mayor simplicidad y ver que se requiere y que se obtiene de cada una: https://trello.com/c/bahvdkrz

jlnunezg10 committed:
Added the functions and routes to create, edit, delete and get goals (single goal and all goals)

jlnunezg10 committed:
added function get_progress_from goal(id)

federico-serron committed:
Merge branch 'development' into goal-backend

federico-serron commented on src/api/routes.py:
```diff
@@ -539,5 +544,153 @@ def delete_category(id):
+        if goal:
+            return(jsonify({'msg':'El nombre de Goal ya existe'})), 404
+        
+        wallets_by_user = Wallet.query.filter_by(user_id = user_id)
```
En la 564 estas buscando si el usuario tiene wallets, no deberia ser si el usuario tiene goals? O quizas estoy entendiendo mal yo

jlnunezg10 replied:
Listo fede, con respecto a esto principalmente tenia la idea de no poder crear un goal si no se tenia un Wallet creado, pero modifique esto borrando esa porcion del codigo y modifique la funcion get_progress_from_wallet() para calcular el progreso sin necesidad de tener una Wallet

federico-serron commented on src/api/routes.py:
```diff
@@ -539,5 +544,153 @@ def delete_category(id):
+def get_all_goals_from_user():
+    try:
+        user_id = get_jwt_identity()
+        goals_from_user = Goal.query.filter_by(user_id = user_id)
```
en la linea 631 te falta agregar `.all()` al final para que te arme una lista con todos los goals, algo asi:

`goals_from_user = Goal.query.filter_by(user_id = user_id).all()`

jlnunezg10 replied:
Agregado el cambio

federico-serron commented on src/api/routes.py:
```diff
@@ -539,5 +544,153 @@ def delete_category(id):
+        db.session.delete(goal_from_user)
+        db.session.commit()
+
+        return (jsonify({'msg': 'Goal eliminada con exito'})) 
```
Falta devolver el codigo `200 `al final

jlnunezg10 replied:
Agregado el cambio

federico-serron commented on src/api/routes.py:
```diff
@@ -539,5 +544,153 @@ def delete_category(id):
+        goal_from_user = Goal.query.filter_by(user_id = user_id, id = id).first()
+
+        if not goal_from_user:
+            return jsonify({'msg': "El goal no existe"})
```
Falta devolver el codigo `404` al final

jlnunezg10 replied:
Agregado el cambio

federico-serron commented on src/api/routes.py:
```diff
@@ -539,5 +544,153 @@ def delete_category(id):
+    try:
+        user_id = get_jwt_identity()
+        goal_from_user = Goal.query.filter_by(user_id = user_id,id=id).first()
+        wallets_by_user = Wallet.query.filter_by(user_id = user_id)
```
EEn la 665 hay que agregar al final el `.all()` para que convierta esa consulta en una lista / array

jlnunezg10 replied:
Agregado el cambio

federico-serron reviewed (changes requested):
Te deje unos comentarios, son detallecitos nomas, cualquier cosa avisame!!

jlnunezg10 committed:
Update routes.py

jlnunezg10 committed:
Update routes.py

Se modifico la funcion get_progress_from_goal() de tal modo que si no existe ninguna wallet, el balance va a ser 0 de tal modo de poder crear un goal sin necesidad de tener un wallet
